src/array/last.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Returns the last element of an array.
+ *
+ * This function takes an array and returns the last element of the array.
+ * If the array is empty, the function returns `undefined`.
+ *
+ * Unlike some implementations, this function is optimized for performance
+ * by directly accessing the last index of the array.
+ *
+ * @param {T[]} arr - The array from which to get the last element.
+ * @returns {T | undefined} The last element of the array, or `undefined` if the array is empty.
+ *
+ * @example
+ * const arr = [1, 2, 3];
+ * const lastElement = last(arr);
+ * // lastElement will be 3
+ *
+ * const emptyArr: number[] = [];
+ * const noElement = last(emptyArr);
+ * // noElement will be undefined
+ */
+
+export function last<T>(arr: readonly [...T[], T]): T;
+export function last<T>(arr: readonly T[]): T | undefined;
+export function last<T>(arr: readonly T[]): T | undefined {
+  if (!Array.isArray(arr)) {
+    throw new Error('not an array');
+  }
```

Comment:
```suggestion
```

I guess `arr` is guaranteed to be an array by types, so we might skip this check.